graphql/database/users/__tests__/deleteUser.test.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-env jest */
+
+import uuid from 'uuid/v4'
+import moment from 'moment'
+
+import UserModel from '../UserModel'
+import deleteUser from '../deleteUser'
+import {
+  DatabaseOperation,
+  getMockUserContext,
+  getMockUserInfo,
+  getMockUserInstance,
+  mockDate,
+  setMockDBResponse,
+  clearAllMockDBResponses,
+} from '../../test-utils'
+import constructFullWidget from '../../widgets/constructFullWidget'
+import BaseWidgetModel from '../../widgets/baseWidget/BaseWidgetModel'
+import UserWidgetModel from '../../widgets/userWidget/UserWidgetModel'
+import getWidgets from '../../widgets/getWidgets'
+
+jest.mock('../../databaseClient')
+jest.mock('../../widgets/getWidgets')
+jest.mock('uuid/v4')
+
+const defaultUserContext = getMockUserContext()
+
+beforeAll(() => {
+  mockDate.on()
+})
+
+afterAll(() => {
+  mockDate.off()
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+  clearAllMockDBResponses()
+})
+
+describe('deleteUser scrubs desired fields: email, username and widgets', () => {
+  it('', async () => {
+    const userInfo = getMockUserInfo()
+    expect.assertions(3)
+
+    // Set mock query responses.
+    const userWidgetsToGet = [
+      new UserWidgetModel({
+        userId: userInfo.id,
+        widgetId: 'abc',
+      }),
+      new UserWidgetModel({
+        userId: userInfo.id,
+        widgetId: 'def',
+      }),
+    ]
+    const baseWidgetsToGet = [
+      new BaseWidgetModel({
+        id: 'abc',
+        position: 2,
+      }),
+      new BaseWidgetModel({
+        id: 'def',
+        position: 1,
+      }),
+    ]
+    const sortedFullWidgets = [
+      constructFullWidget(userWidgetsToGet[1], baseWidgetsToGet[1]),
+      constructFullWidget(userWidgetsToGet[0], baseWidgetsToGet[0]),
+    ]
+    getWidgets.mockResolvedValueOnce(sortedFullWidgets)
+
+    const uuidVal = 'abcd'
```

Comment:
This test is failing because the fake UUID fails schema validation, so use a real UUID here.